lib/domain.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * Domain
+ * @constructor
+ */
+var Domain = function (client) {
+	/**
+	 * Create a new voice call
+	 * @param {Object} params Parameters for creating a new domain
+	 * @param {String} params.name The name is a unique URI to be used in DNS lookups.
+	 * @param {String} params.description String to describe the domain.
+	 * @param {Function} [callback] Callback with the newly created domain
+	 * @return {CallResponse} A promise for the newly created domain
+	 */
+	this.create = function (params, callback) {
+		return client.makeRequest({
+			path   : "domains",
+			method : "POST",
+			body   : params
+		})
+		.then(function (response) {
+			var domain = params;
+			var location = response.headers.location;
+			var domainId = location.substring(location.lastIndexOf("/") + 1);
+			domain.id = domainId;
+			return domain;
+		})
+		.asCallback(callback);
+	};
+
+	/**
+	 * Gets a list of all domains.
+	 * @param {Function} callback A callback with the list of calls
+	 * @return {Promise} A promise for the list of domains.
+	 */
+	this.list = function (callback) {
+		return client.makeRequest({
+			path   : "domains",
+			method : "GET",
+			qs     : { size : 100 }
+		})
+		.then(function (response) {
+			return response.body;
+		})
+		.asCallback(callback);
+	};
+
```

Comment:
For `list` we're building out a `next` link here. Check out how [applications](https://github.com/bandwidthcom/node-bandwidth/blob/2.0-preview/lib/application.js#L27) paginate.

From @fuchsnj 
> We do not want to allow people to use the page param. If someone needs the 2nd page, they should follow the 'next' link.
Using the page parameters has a significant performance impact on Catapult and we are starting to migrate to something else that is faster. If you use the 'next' link we can switch to a different paging method without you knowing.